include/net/rest_client.h

```diff
@@ -0,0 +1,148 @@
+
+	/** In: Socket identifier for the connection. default: REST_CLIENT_SCKT_CONNECT and
+	 *      library will make a new socket connection.
+	 */
+	int connect_socket;
+
+	/** In: If the connection should remain after API call. Default: false. */
+	bool keep_alive;
+
+	/** In: Security tag. Default: REST_CLIENT_SEC_TAG_NO_SEC. */
+	int sec_tag;
+
+	/** In: Indicates the preference for peer verification.
+	 *      Initialize to REST_CLIENT_TLS_DEFAULT_PEER_VERIFY
+	 *      and the default (TLS_PEER_VERIFY_REQUIRED) is used.
+	 */
+	int tls_peer_verify;
+
+	/** In: Used HTTP method. */
+	enum http_method http_method;
+
+	/** In: Hostname or IP address to be used in the request. */
+	const char *host;
+
+	/** In: Port number to be used in the request. */
+	uint16_t port;
+
+	/** In: The URL for this request, for example: /index.html */
+	const char *url;
+
+	/** In: The HTTP header fields. Similar to Zephyr http client.
+	 *      This is a NULL terminated list of header fields. May be NULL.
+	 */
+	const char **header_fields;
+
+	/** In: Payload/body, may be NULL. */
+	const char *body;
+
+	/* Response: **********************************************************/
+
+	/** In: User given timeout value for receiving a response data.
+	 *      Default: CONFIG_REST_CLIENT_REST_REQUEST_TIMEOUT.
+	 */
+	int32_t timeout_ms;
+
+	/** In: User allocated buffer for receiving API response. */
+	char *resp_buff;
+
+	/** In: User given size of resp_buff. */
+	size_t resp_buff_len;
+
+	/** Out: Length of HTTP headers + response body/content data. */
+	size_t total_response_len;
+
+	/** Out: Length of response body/content data. */
+	size_t response_len;
+
+	/** Out: Start of response data (i.e. the body/content) in resp_buff.*/
+	char *response;
+
+	/** Out: Numeric HTTP status code. */
+	uint16_t http_status_code;
+
+	/** Out: Used socket identifier. Use this for keepalived connections as
+	 *       connect_socket for upcoming requests.
+	 */
+	int used_socket_id;
+
+	/** Out: True if used_socket_id was kept alive and wasn't closed after the
+	 *       REST request.
+	 */
+	int used_socket_is_alive;
+};
+
+/**
+ * @brief REST client request.
+ *
+ * @details This function will block until request completes.
+ *
+ * @param[in,out] req_resp_ctx Request and response context for communicating with REST Client API.
+ *
+ * @retval 0, if the REST response has been received successfully. If response_len > 0 then
+ *            there is also body/content data in a response. http_status_code contains the actual
+ *            HTTP status code.
+ *         Otherwise, a (negative) error code is returned.
+ */
+int rest_client_request(struct rest_client_req_resp_context *req_resp_ctx);
+
+/**
+ * @brief Sets the default values into given contexts.
+ *
+ * @details Intended to be used before calling rest_client_request().
+ *
+ * @param[in,out] req_resp_ctx Request and response context for communicating with REST Client API.
```

Comment:
You are not capitalizing the client until now. So I guess it needs to be in lower case.

Reply:
Fixed